cli/tools/registry/diagnostics.rs

```diff
@@ -314,57 +330,65 @@ impl Diagnostic for PublishDiagnostic {
         ]),
       },
-      PublishDiagnostic::InvalidPath { .. } => Cow::Borrowed(&[
+      InvalidPath { .. } => Cow::Borrowed(&[
         Cow::Borrowed("to portably support all platforms, including windows, the allowed characters in package paths are limited"),
       ]),
-      PublishDiagnostic::DuplicatePath { .. } => Cow::Borrowed(&[
+      DuplicatePath { .. } => Cow::Borrowed(&[
         Cow::Borrowed("to support case insensitive file systems, no two package paths may differ only by case"),
       ]),
-      PublishDiagnostic::UnsupportedFileType { .. } => Cow::Borrowed(&[
+      UnsupportedFileType { .. } => Cow::Borrowed(&[
         Cow::Borrowed("only files and directories are supported"),
         Cow::Borrowed("the file was ignored and will not be published")
       ]),
-      PublishDiagnostic::InvalidExternalImport { imported, .. } => Cow::Owned(vec![
+      InvalidExternalImport { imported, .. } => Cow::Owned(vec![
         Cow::Owned(format!("the import was resolved to '{}'", imported)),
         Cow::Borrowed("this specifier is not allowed to be imported on jsr"),
         Cow::Borrowed("jsr only supports importing `jsr:`, `npm:`, and `data:` specifiers"),
       ]),
-      PublishDiagnostic::UnsupportedJsxTsx { .. } => Cow::Owned(vec![
+      UnsupportedJsxTsx { .. } => Cow::Owned(vec![
         Cow::Borrowed("follow https://github.com/jsr-io/jsr/issues/24 for updates"),
+      ]),
+      ExcludedModule { .. } => Cow::Owned(vec![
+        Cow::Borrowed("excluded modules referenced via a package export will error at runtime due to not existing in the package"),
       ])
     }
   }
 
   fn docs_url(&self) -> Option<Cow<'_, str>> {
+    use PublishDiagnostic::*;
     match &self {
-      PublishDiagnostic::FastCheck(diagnostic) => diagnostic.docs_url(),
-      PublishDiagnostic::SpecifierUnfurl(diagnostic) => match diagnostic {
+      FastCheck(diagnostic) => diagnostic.docs_url(),
+      SpecifierUnfurl(diagnostic) => match diagnostic {
         SpecifierUnfurlerDiagnostic::UnanalyzableDynamicImport { .. } => None,
       },
-      PublishDiagnostic::InvalidPath { .. } => {
+      InvalidPath { .. } => {
         Some(Cow::Borrowed("https://jsr.io/go/invalid-path"))
       }
-      PublishDiagnostic::DuplicatePath { .. } => Some(Cow::Borrowed(
+      DuplicatePath { .. } => Some(Cow::Borrowed(
         "https://jsr.io/go/case-insensitive-duplicate-path",
       )),
-      PublishDiagnostic::UnsupportedFileType { .. } => {
+      UnsupportedFileType { .. } => {
         Some(Cow::Borrowed("https://jsr.io/go/unsupported-file-type"))
       }
-      PublishDiagnostic::InvalidExternalImport { .. } => {
+      InvalidExternalImport { .. } => {
         Some(Cow::Borrowed("https://jsr.io/go/invalid-external-import"))
       }
-      PublishDiagnostic::UnsupportedJsxTsx { .. } => None,
+      UnsupportedJsxTsx { .. } => None,
+      ExcludedModule { .. } => {
+        Some(Cow::Borrowed("https://jsr.io/go/excluded-module"))
```

Comment:
todo: add a page for this in the jsr registry

Reply:
Do you want to address this TODO before landing?

Reply:
Yes, if this url looks ok.

Reply:
Looks okay, but it's redirecting to the main doc page right now. I guess we'll create that page after landing this PR?

Reply:
I opened https://github.com/jsr-io/jsr/pull/318